### DISCUSSION TITLE
Import data from prometheus with mimirtool - no metrics found

### DISCUSSION THREAD

**Discussion by appcoders** [grafana/mimir#6437] [Help and support] [answered]
#### Describe the bug
I add old prometheus data via mimirtool backfill. I verified that the tsdb bucket on minio contains the files. Now in Grafana I added datasource http://mimir-nginx.mimir-test.svc:80/prometheus and added Custom Headers: X-Scope-OrgID with value ob for the tenant. But on Explore Grafana only shows "No metrics found".  Why cant' I see the data ? Can I check the Mimir data via cli somehow?

#### To Reproduce

1. Installed mimir-distributed-5.2.0-weekly.259 via helm with compactor_block_upload_enabled in overrides for tentant.
2. `/pvc/prometheus-db # mimirtool backfill --log.level="debug" --address=http://mimir-nginx.mimir-test.svc --id=ob ./01G734T43K1QN8QYP8YJM9F90E` command runs successful.
3. Add datasource to grafana http://mimir-nginx.mimir-test.svc:80/prometheus with X-Scope-OrgID set to ob
4. Check Datasource in Grafana with Explore

#### Expected behavior

Successfully explore the metrics of the imported Prometheus data in Grafana.

#### Environment
Infrastructure: Kubernetes
Deployment tool: helm

**Comment by appcoders** [comment]
Patience is the key. After some hours, the metrics materialized. Is there a way to monitor the progress of processing state?

**Reply by pstibrany** [reply]
There can be several reasons for this:

- compactor needs to discover new blocks and include them in bucket-index. This happens every 15 minutes by default.
- querier needs to load updated bucket-index to be aware of the blocks. This happens every 15 minutes by default.
- store-gateway also needs to load updated bucket-index (every 15m by default), and then load the index header of new blocks. This can take a while.
- results cache may contain results before the blocks were available
- if backfilled blocks are too new, they may not be queryable until later.

Last two points are discussed at https://grafana.com/docs/mimir/latest/configure/configure-tsdb-block-upload/#the-results-cache-needs-flushing

**Reply by pstibrany** [reply, marked as answer]
Compactor exports metrics with blocks count that can be used to monitor when bucket-index was updated.

Store-gateway exports metric with loaded blocks count.
